docs/spec_draft.md

```diff
@@ -1408,7 +1487,9 @@ where `i[d] = j[permutation[d]]`.
   * (C1) `operand` and `result` have the same element type.
   * (C2) `permutation` is a permutation of `[0, 1, ..., R-1]` where `R` is the
   rank of `operand`.
-  * (C3) `result`'s shape is a permutation of `operand`'s shape.
+  * (C3) `result`'s shape is a permutation of `operand`'s shape w.r.t
+  `permutation`. Formally, for all dimensions `i` in `operand`,
+  `dim(operand, i) = dim(result, j)` where `j = permutation[i]`.
```

Comment:
How about we drop the informal part? I.e. only keep "for all dimensions `i` in `operand`,`dim(operand, i) = dim(result, j)` where `j = permutation[i]`".